cartography/intel/azure/sql.py

```diff
@@ -277,267 +291,237 @@ def load_server_details(neo4j_session, credentials, subscription_id, details, up
 
 
 @timeit
-def _load_server_dns_aliases(neo4j_session, dns_aliases, update_tag):
+def _load_server_dns_aliases(
+        neo4j_session: neo4j.Session, dns_aliases: List[Dict], update_tag: int
+) -> None:
     """
     Ingest the DNS Alias details into neo4j.
     """
     ingest_dns_aliases = """
-    MERGE (alias:AzureServerDNSAlias{id: {DNSAliasId}})
+    UNWIND {dns_aliases_list} as dns_alias
+    MERGE (alias:AzureServerDNSAlias{id: dns_alias.id})
     ON CREATE SET alias.firstseen = timestamp(), alias.lastupdated = {azure_update_tag}
-    SET alias.name = {Name},
-    alias.dnsrecord = {AzureDNSRecord}
-    WITH alias
-    MATCH (s:AzureServer{id: {ServerId}})
+    SET alias.name = dns_alias.name,
+    alias.dnsrecord = dns_alias.azure_dns_record
+    WITH alias, dns_alias
+    MATCH (s:AzureSQLServer{id: dns_alias.server_id})
     MERGE (s)-[r:USED_BY]->(alias)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {azure_update_tag}
     """
 
-    for data in dns_aliases:
-        neo4j_session.run(
-            ingest_dns_aliases,
-            DNSAliasId=data['id'],
-            Name=data['name'],
-            AzureDNSRecord=get_optional_value(data, ['azure_dns_record']),
-            ServerId=data['server_id'],
-            azure_update_tag=update_tag,
-        )
+    neo4j_session.run(
+        ingest_dns_aliases,
+        dns_aliases_list=dns_aliases,
+        azure_update_tag=update_tag,
+    )
 
 
 @timeit
-def _load_server_ad_admins(neo4j_session, ad_admins, update_tag):
+def _load_server_ad_admins(
+        neo4j_session: neo4j.Session, ad_admins: List[Dict], update_tag: int
+) -> None:
     """
     Ingest the Server AD Administrators details into neo4j.
     """
     ingest_ad_admins = """
-    MERGE (a:AzureServerADAdministrator{id: {AdAdminId}})
+    UNWIND {ad_admins_list} as ad_admins
+    MERGE (a:AzureServerADAdministrator{id: ad_admins.id})
     ON CREATE SET a.firstseen = timestamp(), a.lastupdated = {azure_update_tag}
```

Comment:
lastupdated should not be in the create set. it should be in update set.

Reply:
done